### PR TITLE
les: fix private net issues, enable adding peers manually again

### PR DESCRIPTION
**Pull request by zsfelfoldi**
This PR solves https://github.com/ethereum/go-ethereum/issues/3510

- enables adding peers manually by adding new incoming connections to the server pool instead of rejecting everything not initiated by it
- removes the mechanism which delayed starting the LES server side protocol until ETH was fully synced. This mechanism solved a problem that does not exist anymore, clients now check server's reported head before sending requests and choose a server that can actually serve it. At the same time it caused other problems, like not starting LES server at all when the node was mining alone on a private net. Also, not starting the protocol stalled connections to other LES servers through ETH, even though they did not want to do anything with each other through LES.
- allows disabling topic discovery with --nodiscover option

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -663,7 +663,7 @@ func MakeNode(ctx *cli.Context, name, gitCommit string) *node.Node {
 		Version:           vsn,
 		UserIdent:         makeNodeUserIdent(ctx),
 		NoDiscovery:       ctx.GlobalBool(NoDiscoverFlag.Name) || ctx.GlobalBool(LightModeFlag.Name),
-		DiscoveryV5:       ctx.GlobalBool(DiscoveryV5Flag.Name) || ctx.GlobalBool(LightModeFlag.Name) || ctx.GlobalInt(LightServFlag.Name) > 0,
+		DiscoveryV5:       ctx.GlobalBool(DiscoveryV5Flag.Name) || (ctx.GlobalBool(LightModeFlag.Name) || ctx.GlobalInt(LightServFlag.Name) > 0) && !ctx.GlobalBool(NoDiscoverFlag.Name),
```
Comment (holiman) [inline cmd/utils/flags.go]: I find this hard to parse for a human brain. Is it possible to reconstruct this statement so it's easier to brain-validate e.g. what effect the `NoDiscoverFlag` has on the outcome?

File: eth/handler.go [hunk 2]
```diff
@@ -173,7 +173,7 @@ func NewProtocolManager(config *params.ChainConfig, fastSync bool, networkId int
 		return blockchain.CurrentBlock().NumberU64()
 	}
 	inserter := func(blocks types.Blocks) (int, error) {
-		manager.setSynced() // Mark initial sync done on any fetcher import
+		atomic.StoreUint32(&manager.synced, 1) // Mark initial sync done on any fetcher import
```
Comment (holiman) [inline eth/handler.go]: It doesn't hurt to use `atomic` here, but I'm curious... Previously, you actually _did_ something if `synched` went from `0` to `1`, and thus wanted it to be threadsafe/atomic. Now it seems you could just aswell use non-atomic, right?

Comment (zsfelfoldi) [inline eth/handler.go]: No, synced is being read and written in different goroutines, see:
https://github.com/ethereum/go-ethereum/blob/master/eth/handler.go#L663
This is how it worked before I added setSynced (I restored it in its original state).

File: cmd/utils/flags.go (hunk 1 above)
Comment (zsfelfoldi) [inline cmd/utils/flags.go]: you're right, I shouldn't write expressions like this because it was not exactly what I intended to do :) it's now fixed, reconstructed and clarified.
